### ISSUE TITLE
[Bug] segmentation fault in debug mode, when initializing core

### ISSUE THREAD

**Issue by zhibo1-wang** [openvinotoolkit/openvino#8698]
##### System information (version)
- OpenVINO => 2021.4.582
- Operating System / Platform => Windows 10 21H2
- Compiler => Visual Studio 2019
- Problem classification => General

##### Detailed description
**In Release build, everything works fine**, but
In Debug build, any function calling OpenVINO API that use a std::string as parameter, like initialize Core with a std::string, core.RegisterPlugins, or core.ReadModel, will cause a segmentation fault. 

![image](https://user-images.githubusercontent.com/89496158/142558312-47407633-426e-4b75-b910-2edd95a3f189.png)

I inspected variables inside OpenVINO's library in debug mode, the data in the std::string instance is corrupted. 

![image](https://user-images.githubusercontent.com/89496158/142558430-071381c0-043e-4433-8282-a3c8a62a1133.png)

Outside the ie_core code the string is good.

![image](https://user-images.githubusercontent.com/89496158/142558642-d4cfd3c1-d334-453e-8f3c-7e938d833418.png)

Firstly I suspect it is a mismatch of C++ STL ABI, however the release mode works fine. I am stuck and cannot find anything that helps.

Please help me make the Core successfully initialized with a std::string parameter, or tell me where my usage of OpenVINO is wrong.

##### Steps to reproduce
The code belongs to another project, which is built by ninja build system. The entry of the failed scenario is at
https://github.com/open-webrtc-toolkit/owt-client-native/pull/564/files#diff-9e43547b39a6f1ad6f6e9bac2536360b63e7345788d015f958e16d4f63a9c118
The program expects a model path, but you can use any value as it fails before reading the model.

In short, I use openvino in my shared library, via dynamic linking. And I dynamically loaded the shared library in my test program.
(1) Main program ==dynamic==> (2) My DLL ==dynamic==> (3) OpenVINO
It fails when (2) calls (3).

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by zhibo1-wang** [comment]
Solved by using `-D_ITERATOR_DEBUG_LEVEL=2` and `/MDd`. For anyone having issue only in debug build, try them.

**Comment by zhibo1-wang** [comment]
Besides, this could be mentioned in the developers' document.
